pkg/api/store/nodetemplate/store.go

```diff
@@ -24,43 +24,95 @@ type nodeTemplateStore struct {
 	types.Store
 	NodePoolLister        v3.NodePoolLister
 	CloudCredentialLister corev1.SecretLister
-	NodeTemplateClient    v3.NodeTemplateInterface
+}
+
+type transformer struct {
+	NodeTemplateClient v3.NodeTemplateInterface
 }
 
 func Wrap(store types.Store, npLister v3.NodePoolLister, secretLister corev1.SecretLister, ntClient v3.NodeTemplateInterface) types.Store {
-	s := &nodeTemplateStore{
-		Store:                 store,
+	t := &transformer{
+		NodeTemplateClient: ntClient,
+	}
+	s := &transform.Store{
+		Store:       store,
+		Transformer: t.filterLegacyTemplates,
+	}
+	return &nodeTemplateStore{
+		Store:                 s,
 		NodePoolLister:        npLister,
 		CloudCredentialLister: secretLister,
-		NodeTemplateClient:    ntClient,
-	}
-	return &transform.Store{
-		Store:       s,
-		Transformer: s.filterLegacyTemplates,
 	}
 }
 
-func (s *nodeTemplateStore) filterLegacyTemplates(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
+func (t *transformer) filterLegacyTemplates(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
 	ns, _ := data["namespaceId"].(string)
-	name, _ := data["name"].(string)
+	id, _ := data["id"].(string)
```

Comment:
can you pls explain why is this change needed?

Reply:
name was the display name, so nothing which is not valid for enqueueing